packages/firebase_storage/firebase_storage/ios/Classes/FLTFirebaseStoragePlugin.m

```diff
@@ -227,136 +321,256 @@ - (void)referenceGetDownloadUrl:(id)arguments
+                         reference:(PigeonStorageReference *)reference
+                          metadata:(PigeonSettableMetadata *)metadata
+                        completion:(void (^)(PigeonFullMetaData *_Nullable,
+                                             FlutterError *_Nullable))completion {
+  FIRStorageReference *storage_reference = [self getFIRStorageReferenceFromPigeon:app
+                                                                        reference:reference];
+  FIRStorageMetadata *storage_metadata = [self getFIRStorageMetadataFromPigeon:metadata];
+
+  [storage_reference updateMetadata:storage_metadata
+                         completion:^(FIRStorageMetadata *updatedMetadata, NSError *error) {
+                           if (error != nil) {
+                             completion(nil, [self FlutterErrorFromNSError:error]);
+                           } else {
+                             NSDictionary *dict = [FLTFirebaseStoragePlugin
+                                 NSDictionaryFromFIRStorageMetadata:updatedMetadata];
+                             completion([PigeonFullMetaData makeWithMetadata:dict], nil);
+                           }
+                         }];
 }
 
-- (void)taskStartPutString:(id)arguments
-      withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
-  [self startFIRStorageObservableTaskForArguments:arguments
-                    andFLTFirebaseStorageTaskType:FLTFirebaseStorageTaskTypeString];
-  result.success(nil);
+- (void)taskPauseApp:(PigeonStorageFirebaseApp *)app
+              handle:(NSNumber *)handle
+          completion:(void (^)(NSDictionary<NSString *, id> *_Nullable,
+                               FlutterError *_Nullable))completion {
+  // TODO
+  NSLog(@"taskPauseApp");
 }
 
-- (void)taskStartPutFile:(id)arguments withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
-  [self startFIRStorageObservableTaskForArguments:arguments
-                    andFLTFirebaseStorageTaskType:FLTFirebaseStorageTaskTypeFile];
-  result.success(nil);
+- (void)taskResumeApp:(PigeonStorageFirebaseApp *)app
+               handle:(NSNumber *)handle
+           completion:(void (^)(NSDictionary<NSString *, id> *_Nullable,
+                                FlutterError *_Nullable))completion {
+  // TODO
+  NSLog(@"taskResumeApp");
 }
 
-- (void)taskWriteToFile:(id)arguments withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
-  [self startFIRStorageObservableTaskForArguments:arguments
-                    andFLTFirebaseStorageTaskType:FLTFirebaseStorageTaskTypeDownload];
-  result.success(nil);
+- (void)taskCancelApp:(PigeonStorageFirebaseApp *)app
+               handle:(NSNumber *)handle
+           completion:(void (^)(NSDictionary<NSString *, id> *_Nullable,
+                                FlutterError *_Nullable))completion {
+  // TODO
+  NSLog(@"taskCancelApp");
 }
 
 - (void)taskPause:(id)arguments withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
```

Comment:
`taskPause`, `taskCancel` & `taskResume` are the previous implementation and should be replaced with the above noted methods used by generated Pigeon code.

Reply:
done